modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -26,9 +26,19 @@ const findUrlEnd = (characters: string[], startIndex: number): number => {
   return peakedWord.substr(0, 3) === '://' ? endIndex : startIndex;
 };
 
-const findWords = <T>(chars: T[], sChars: string[], characterMap: CharacterMap, options: WordOptions): T[][] => {
-  const words: T[][] = [];
-  let word: T[] = [];
+export type Word<T> = T[];
+interface WordIndex {
+  readonly start: number;
+  readonly end: number;
+}
+export interface WordsWithIndices<T> {
+  readonly words: Word<T>[];
+  readonly indices: WordIndex[];
+}
+const findWordsWithIndices = <T>(chars: Word<T>, sChars: string[], characterMap: CharacterMap, options: WordOptions): WordsWithIndices<T> => {
```

Comment:
I'm told to not be picky about formatting but maybe space this up a bit.

```suggestion
export type Word<T> = T[];

interface WordIndex {
  readonly start: number;
  readonly end: number;
}

export interface WordsWithIndices<T> {
  readonly words: Word<T>[];
  readonly indices: WordIndex[];
}

const findWordsWithIndices = <T>(chars: Word<T>, sChars: string[], characterMap: CharacterMap, options: WordOptions): WordsWithIndices<T> => {
```